designdocs/passkey_p256.md

```diff
@@ -0,0 +1,556 @@
+
+    ```typescript
+
+    interface PasskeyPayload {
+      passkeyPublicKey: `passkey_pk`;
+      passkeySignature: `passkey_sig_tx`;
+      passkeyAuthenticator: `passkey_authenticator_data`;
+      passkeyClientDataJson: `passkey_client_data`;
+      passkeyCall: PasskeyCall;
+    }
+
+    interface PasskeyCall {
+      accountId: `account_pk`;
+      accountOwnershipProof: `AccountOwnershipProof`;
+      nonce: `nonce`;
+      call: Call;
+    }
+
+    interface AccountOwnershipProof {
+      Sr25519: account_sig_passkey_pk;
+    }
+    ```
+
+- The `PasskeyPayload` contains the following fields:
+- `passkey_pk` : Passkey Public Key
+- `passkeySignature` : Signature generated by Passkey on transaction payload
+- `passkeyAuthenticator` : Authenticator data generated by Passkey
+- `passkeyClientDataJson` : Client data generated by Passkey
+- `passkeyCall` : The actual transaction call that needs to be submitted on-chain.
+- `AccountOwnershipProof` : A proof of account ownership generated by the Account Key. This is used to prove ownership of the account.
+- `nonce` : A unique number used to prevent replay attacks.
+
+### Passkey Registration
+
+1. **User visits "Wallet Setup Page"**:
+   - The browser JavaScript generates a seed phrase, using it to generate `account_keypair` and the corresponding account public key (`account_pk`).
+
+2. **Web Wallet requests creation of user Passkey**:
+   - A challenge containing the `account_pk` is sent to the Passkey generation process.
+
+3. **Browser JavaScript handles Passkey creation**:
+   - The Passkey generation process creates a Passkey and provides the `passkey_pk`.
+   - The Passkey signs the `account_pk`, resulting in `passkey_sig_account_pk`.
+
+4. **Browser JavaScript generates a signature using the seed phrase**:
+   - `account_keypair` is used to generate a signature on the `passkey_pk`, resulting in `account_sig_passkey_pk`.
+
+      ```javascript
+        // Generate Account Key Pair
+        const mnemonic = "GENERATE SOME MNEMONIC HERE"
+        const keyring = new Keyring({ type: 'sr25519' });
+        const account_keypair = keyring.fromMnemonic(mnemonic);
+        ...
+        ...
+        const account_pk = account_keypair.publicKey;
+
+        // Generate Passkey and Sign Account Public Key
+        const options = generateRegistrationOptions();
+        options.challenge = account_pk;
+        const registerPassKey = await SimpleWebAuthnBrowser.startRegistration(options);
+        ...
+        ...
+        // Sign Passkey Public Key using Account Key
+        const passkey_sig_account_pk = registerPassKey.signature; // obtained from the registration response
+        const passkey_pk = registerPassKey.publicKey; // obtained from the registration response
+
+        // Sign Passkey Public Key using Account Key, this is used to prove ownership of the account
+        const account_sig_passkey_pk = account_keypair.sign(passkey_pk);
+      ```
```

Comment:
Perhaps either:
(a) interpolate code block in the description of each step, ie:
1.  **User visits Wallet Setup Page**
(code block)

2. etc...
(code block)


---or---
keep the entire code block where it is, but number the comments in the block to match the descriptions above